### PR TITLE
PDF from URL with download progress status (with updated example)

### PR DESCRIPTION
**Pull request by abhisheksrocks**
**TLDR**
New method to load pdf from url with current download status

**Why?**
If the PDF file is large, or the internet is slow, it would be good for the user to know the how much(or how fast) the file is downloading.

**What**
New function in PDFDocument with callbacks for download progress status. Also updated the example's main.dart to demonstrate this feature.

**How**
Using FlutterCacheManger, invoking getFileStream() instead of the getSingleFile() method to download the file. We then make use of the Stream\<FileResponse\> to invoke the downloadProgress (continuously while downloading) and onDownloadComplete(once download is finished) callbacks in our created function.

Using this introduced method we can get the download information, which can be used like so:

Screenshot 2022-08-27 at 5 52 18 AM

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -1,11 +1,101 @@
...
+  }
+
+  Widget buildProgress() {
+    if (downloadProgress == null) return SizedBox();
+
+    String parseBytesToKBs(int bytes) {
+      return '${(bytes / 1000).toStringAsFixed(2)} KBs';
+    }
+
+    String progressString =
```
Comment (kaichii) [inline example/lib/main.dart:63]: Double `KBs`, maybe we should remove one.

Comment (abhisheksrocks) [reply example/lib/main.dart:63]: Oh missed it! 😄
